### ISSUE TITLE
Check that the SQL script we get from EF migration is the same as the scripts we now use

### ISSUE THREAD

**Issue by SandGrainOne** [Altinn/altinn-profile#234]
### Description

We want to run database migration with the help of Entity Framework. The scripts being generated by EF needs to create exactly the same database definitions as those we already have. Any differences must be corrected.

### Additional Information

It could be that the scripts we currently have were initially made by EF and that there are significantly fewer differences than we fear.

Commands:

```
dotnet ef migrations add InitialCreate --project Altinn.Profile.Integrations.csproj --startup-project ..\Altinn.Profile\Altinn.Profile.csproj
dotnet ef migrations script --project Altinn.Profile.Integrations.csproj --startup-project ..\Altinn.Profile\Altinn.Profile.csproj
```

This will require the EF design package. It can be added to the Integrations project and will be available for the main project if the PrivateAssets tag/property is removed from the PackageReference in the integration project.

### Tasks

- [x] Add necessary package reference to the integration project and make assets available for the startup project.
- [x] Run the commands to initialize migration state and generate skripts
- [x] Compare the result with existing scripts
- [x] Make adjustments as needed. Either in the defined entities used by EF or in the Migration scripts. Differences can be discussed by the team.

### Analysis
**TLDR**:
Entity frameworks creates unique indexes instead of column constraints. pgAdmin use constraints as expected
Type names: 
- INT vs integer 
- CHAR(\*) vs character(\*)
- TIMESTAMPTZ vs timestamp with time zone
- VARCHAR(\*) vs character varying(\*)

When it's about types it's interesting to see that the names being used by EF is the same as those I get when I ask pgAdmin to generate scripts for me.

---
Table definition for `contact_and_reservation.mailbox_supplier`
Current script:
```
CREATE TABLE IF NOT EXISTS contact_and_reservation.mailbox_supplier (
    mailbox_supplier_id INT GENERATED ALWAYS AS IDENTITY (START WITH 1 INCREMENT BY 1) PRIMARY KEY,
    org_number_ak CHAR(9) NOT NULL,
    CONSTRAINT unique_org_number_ak UNIQUE (org_number_ak)
);
```
EF generated:
```
CREATE TABLE contact_and_reservation.mailbox_supplier (
    mailbox_supplier_id integer GENERATED ALWAYS AS IDENTITY,
    org_number_ak character(9) NOT NULL,
    CONSTRAINT mailbox_supplier_pkey PRIMARY KEY (mailbox_supplier_id)
);
CREATE UNIQUE INDEX unique_org_number_ak ON contact_and_reservation.mailbox_supplier (org_number_ak);
```

Table definition for `contact_and_reservation.metadata`
Current script:
```
CREATE TABLE IF NOT EXISTS contact_and_reservation.metadata (
    latest_change_number BIGINT PRIMARY KEY,
    exported TIMESTAMPTZ
);
```
EF generated:
```
CREATE TABLE contact_and_reservation.metadata (
    latest_change_number bigint NOT NULL,
    exported timestamp with time zone,
    CONSTRAINT metadata_pkey PRIMARY KEY (latest_change_number)
);
```

Table definition for `contact_and_reservation.person`
Current script:
```
CREATE TABLE IF NOT EXISTS contact_and_reservation.person (
    contact_and_reservation_user_id INT GENERATED ALWAYS AS IDENTITY (START WITH 1 INCREMENT BY 1) PRIMARY KEY,
    fnumber_ak CHAR(11) NOT NULL UNIQUE,
    reservation BOOLEAN,
    description VARCHAR(20),
    mobile_phone_number VARCHAR(20),
    mobile_phone_number_last_updated TIMESTAMPTZ,
    mobile_phone_number_last_verified TIMESTAMPTZ,
    email_address VARCHAR(400),
    email_address_last_updated TIMESTAMPTZ,
    email_address_last_verified TIMESTAMPTZ,
    mailbox_address VARCHAR(50),
    mailbox_supplier_id_fk INT,
    x509_certificate TEXT,
    language_code CHAR(2) NULL,
    CONSTRAINT fk_mailbox_supplier FOREIGN KEY (mailbox_supplier_id_fk) REFERENCES contact_and_reservation.mailbox_supplier (mailbox_supplier_id),
    CONSTRAINT chk_language_code CHECK (language_code ~* '^[a-z]{2}$')
);
CREATE INDEX idx_fnumber_ak ON contact_and_reservation.person (fnumber_ak);
```
EF generated:
```
CREATE TABLE contact_and_reservation.person (
    contact_and_reservation_user_id integer GENERATED ALWAYS AS IDENTITY,
    fnumber_ak character(11) NOT NULL,
    reservation boolean,
    description character varying(20),
    mobile_phone_number character varying(20),
    mobile_phone_number_last_updated timestamp with time zone,
    mobile_phone_number_last_verified timestamp with time zone,
    email_address character varying(400),
    email_address_last_updated timestamp with time zone,
    email_address_last_verified timestamp with time zone,
    mailbox_address character varying(50),
    mailbox_supplier_id_fk integer,
    x509_certificate text,
    language_code character(2),
    CONSTRAINT person_pkey PRIMARY KEY (contact_and_reservation_user_id),
    CONSTRAINT fk_mailbox_supplier FOREIGN KEY (mailbox_supplier_id_fk) REFERENCES contact_and_reservation.mailbox_supplier (mailbox_supplier_id)
);
CREATE INDEX idx_fnumber_ak ON contact_and_reservation.person (fnumber_ak);
CREATE INDEX "IX_person_mailbox_supplier_id_fk" ON contact_and_reservation.person (mailbox_supplier_id_fk);
CREATE UNIQUE INDEX person_fnumber_ak_key ON contact_and_reservation.person (fnumber_ak);
```
---

### Acceptance Criterias

- [ ] The generated scripts produce exactly the same result as the existing scripts in the Migration folder.
